samcli/lib/providers/sam_function_provider.py

```diff
@@ -503,7 +551,93 @@ def _parse_layer_info(
                     str(layer),
                 )
 
-        return layers
+        return layers + found_layers
+
+    @staticmethod
+    def _search_layer(
```

Comment:
Right now this method does not support more complicated intrinsics like Fn:Sub and Fn:If, let's mention this in the update and have a future task to expand support. One of possible solutions can be adding in an intrinsic resovler.